GooglePlayInstant/Editor/BundletoolDownloadWindow.cs

```diff
@@ -0,0 +1,171 @@
+                    _downloadRequest = null;
+
+                    Debug.LogErrorFormat("Bundletool download error: {0}", downloadRequestError);
+                    if (EditorUtility.DisplayDialog("Download Failed",
+                        downloadRequestError + "\n\nClick \"OK\" to retry.", "OK", "Cancel"))
+                    {
+                        StartDownload();
+                    }
+                    else
+                    {
+                        Close();
+                    }
+
+                    return;
+                }
+
+                // Download succeeded. Copy the bytes if this version of Unity doesn't support DownloadHandlerFile.
+                var bundletoolJarPath = Bundletool.GetBundletoolJarPath();
+#if !UNITY_2017_2_OR_NEWER
+                File.WriteAllBytes(bundletoolJarPath, _downloadRequest.downloadHandler.data);
+#endif
+                _downloadRequest.Dispose();
+                _downloadRequest = null;
+
+                Debug.LogFormat("Bundletool downloaded: {0}", bundletoolJarPath);
+                var message = string.Format(
+                    "Bundletool has been downloaded to your project's \"Library\" directory: {0}", bundletoolJarPath);
+                if (EditorUtility.DisplayDialog("Download Complete", message, "OK"))
+                {
+                    Close();
+                }
+
+                return;
+            }
+
+            // Download is in progress.
+            if (EditorUtility.DisplayCancelableProgressBar(
+                "Downloading bundletool", null, _downloadRequest.downloadProgress))
+            {
+                EditorUtility.ClearProgressBar();
+                _downloadRequest.Abort();
+                _downloadRequest.Dispose();
+                _downloadRequest = null;
+                Debug.Log("Cancelled bundletool download.");
+            }
+        }
+
+        private void OnDestroy()
+        {
+            if (_downloadRequest != null)
+            {
+                _downloadRequest.Dispose();
+                _downloadRequest = null;
+            }
+        }
+
+        private void StartDownload()
+        {
+            Debug.Log("Downloading bundletool...");
+            var bundletoolUri = string.Format(
+                "https://github.com/google/bundletool/releases/download/{0}/bundletool-all-{0}.jar",
+                Bundletool.BundletoolVersion);
+#if UNITY_2017_2_OR_NEWER
+            var downloadHandler = new DownloadHandlerFile(Bundletool.GetBundletoolJarPath())
+            {
+                removeFileOnAbort = true
+            };
+            _downloadRequest = new UnityWebRequest(bundletoolUri, UnityWebRequest.kHttpVerbGET, downloadHandler, null);
+            _downloadRequest.SendWebRequest();
+#else
+            _downloadRequest = UnityWebRequest.Get(bundletoolUri);
+            _downloadRequest.Send();
+#endif
+        }
+
+        private static void CreateButton(string text, Action action)
```

Comment:
This could be broken out into a WindowUtils class. There's a handful of other places that could also use that function (PlayerSettingsWindow and QuickDeployWindow)

Reply:
Created WindowUtils. Currently there is nothing else but this class that was able to use it, but the method seems like something we'll use later for QuickDeployWindow. And the class seems useful to have around going forward.